src/pages/Tasks/Tasks.jsx

```diff
@@ -0,0 +1,392 @@
+    this.addUserHandler = this.addUserToTask.bind(this);
+    this.onShowDetails = this.showTaskDetails.bind(this);
+  }
+
+  async componentDidMount() {
+    await this.getTasks();
+    await this.getUsers();
+  }
+
+  async getTasks() {
+    const data = await getTasksData();
+    this.setState({
+      data,
+    });
+  }
+
+  async getUsers() {
+    const usersData = await getUsersData();
+    this.setState({
+      usersData,
+    });
+  }
+
+  async getUserTasks() {
+    const userTasks = await getUserTasks();
+    this.setState({
+      userTasks,
+    });
+  }
+
+  getUserTasksWithActiveTask() {
+    const { userTasks, editTaskData, readTaskData } = this.state;
+    if (editTaskData || readTaskData) {
+      const taskId = editTaskData ? editTaskData.id : readTaskData.id;
+      const userTasksWithActiveTask = userTasks
+        ? userTasks
+            .filter((userTask) => {
+              return userTask.taskId === taskId;
+            })
+            .map((userTask) => {
+              return {
+                ...userTask,
+                checked: true,
+              };
+            })
+        : userTasks;
+      this.setState({
+        userTasksWithActiveTask,
+      });
+    }
+
+    return null;
+  }
+
+  async openCreateModal(event, editTaskData = null) {
+    if (editTaskData) {
+      this.setState({
+        modalIsOpen: true,
+        editTaskData,
+      });
+    } else {
+      this.setState({
+        modalIsOpen: true,
+      });
+    }
+    await this.getUserTasks();
+    this.getUserTasksWithActiveTask();
+  }
+
+  closeCreateModal() {
+    this.setState({
+      modalIsOpen: false,
+      editTaskData: null,
+      userTasksWithActiveTask: null,
+      readTaskData: null,
+      readOnly: false,
+    });
+  }
+
+  async addNewTask(taskData) {
+    const { editTaskData } = this.state;
+    const { startDate, deadlineDate } = taskData;
+    const formatedTaskData = {
+      ...taskData,
+      startDate: new Date(startDate).getTime(),
+      deadlineDate: new Date(deadlineDate).getTime(),
+    };
+    let { userTasksWithActiveTask } = this.state;
+    let taskId = null;
+    if (editTaskData) {
+      taskId = editTaskData.id;
+      await writeNewTaskData(taskId, formatedTaskData);
+    } else {
+      taskId = await writeNewTaskData(null, formatedTaskData);
+      userTasksWithActiveTask = userTasksWithActiveTask.map((userTask) => {
+        return {
+          ...userTask,
+          taskId,
+        };
+      });
+    }
+    await updateUserTasks(userTasksWithActiveTask);
+    await this.getTasks();
+    this.closeCreateModal();
+  }
+
+  editTask(editTaskData) {
+    this.openCreateModal(null, editTaskData);
+  }
+
+  async deleteTask(taskId) {
+    await deleteTask(taskId);
+    this.getTasks();
+  }
+
+  addUserToTask(userId, taskId, userTaskId, isChecked) {
+    const { userTasksWithActiveTask } = this.state;
+    let newUsersTaskData = userTasksWithActiveTask ? [...userTasksWithActiveTask] : [];
```

Comment:
userTasksWithActiveTask

Reply:
I tried to rename it.
```suggestion
  addUserToTask(userId, taskId, userTaskId, isChecked) {
    const { userTasksWithCurrentTaskId } = this.state;
```